playbook/app/pb_kits/playbook/pb_filter/Filter/index.tsx

```diff
@@ -1,33 +1,59 @@
-import React from 'react'
-import FilterSingle, { FilterSingleProps } from './FilterSingle'
-import FilterDouble, { FilterDoubleProps } from './FilterDouble'
+import React, { useState } from 'react';
+import FilterSingle, { FilterSingleProps } from './FilterSingle';
+import FilterDouble, { FilterDoubleProps } from './FilterDouble';
 
 type FilterProps =
   | FilterSingleProps
   | (FilterDoubleProps & {
-      double?: boolean,
-    })
+      double?: boolean;
+    });
 
 const Filter = ({
   double = false,
   ...templateProps
-  }: FilterProps): React.ReactElement => {
+}: FilterProps): React.ReactElement => {
+  const [selectedDate, setSelectedDate] = useState<string | null>(null);
+
+  const handleDateClick = (child: Element) => {
+    const dates = child.textContent;
+    setSelectedDate(dates);
+    console.log('Selected date:', dates);
+    
+    const datePickers = Array.from(document.getElementsByClassName('quick-pick-ul')[0].children);
+    datePickers.forEach((el) => el.classList.remove('selected'));
+    child.classList.add('selected');
+  };
+
+  const handlePopoverOpen = () => {
+    const datePickers = Array.from(document.getElementsByClassName('quick-pick-ul')[0].children);
+    
+    datePickers.forEach((child: Element) => {
+      child.removeEventListener('click', handleDateClick);
+      
+      child.addEventListener('click', () => handleDateClick(child));
+    });
+  };
+
```

Comment:
This is where im grabbing what is selected on the date picker kit and setting state with the value